common/utils/retry_test.go

```diff
@@ -0,0 +1,93 @@
+package utils
+
+import (
+	"math/rand"
+	"testing"
+	"time"
+
+	"github.com/quintilesims/layer0/common/errors"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestRetryTimeout(t *testing.T) {
+	start := time.Now()
+	fn := func() (bool, error) {
+		time.Sleep((time.Duration(rand.Intn(2))) * time.Millisecond)
+		return true, nil
+	}
+
+	if err := Retry(time.Millisecond*1, time.Millisecond*1, fn); err != nil {
+		assert.WithinDuration(t, start.Add(time.Millisecond*1), time.Now(), time.Millisecond)
+	}
+}
+
+func TestRetryNoTimeout(t *testing.T) {
```

Comment:
This behavior doesn't make much sense. Looking at the test itself, it looks like you're actually testing that fn is repeatedly called while you return true, nil